mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/FiltersDialog.kt

```diff
@@ -0,0 +1,143 @@
+package ch.epfl.sdp.mobile.ui.tournaments
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.layout.Arrangement.spacedBy
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Alignment.Companion.CenterVertically
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import ch.epfl.sdp.mobile.state.LocalLocalizedStrings
+import ch.epfl.sdp.mobile.ui.PawniesColors
+
+/** An interface representing the state of the [FiltersDialog] composable. */
+@Stable
```

Comment:
Question: Why is there a `@Stable` Annotation ?

Reply:
What was your thought behind it?

Reply:
The `@Stable` annotation is a contract with the Compose compiler which helps it generate some more optimised code. Long story short, when annotating an interface with `@Stable`, we must make sure all the external properties of the interface implementations will be observable by the composition (aka they use the Snapshot system).

All our viewmodel interfaces have `@Stable` guarantees.

(ref : https://developer.android.com/reference/kotlin/androidx/compose/runtime/Stable)